:sparkles: Add Verbose Argument

edelciomolina opened the pull request:
Add ability to use the 'verbose' argument that allows printing stdout and stderr in real-time, without needing to collect the output after the entire execution.

![image](https://github.com/user-attachments/assets/3ea293c6-c5f9-4b2f-9d98-3842f03d995d)

edelciomolina committed:
Add Verbose Argument

jdexyz commented on index.js:
```diff
@@ -70,14 +71,29 @@ const dump = function ({
+    console.log(data.toString().trim());
+  });
+
+  subprocess.stderr.on('data', (data) => {
+    console.info(data.toString().trim());
+  });
```
Why console.info and not console.error ?

edelciomolina replied:
I had previously used `.error`, but I noticed that pg_dump sometimes prints warnings as stderr, and when that happened, the dump process would stop halfway. Let me work on an improvement here, and I'll provide an update on the PR shortly.

jdexyz replied:
In your other PRs I saw you created a more complete checkError function. So once everything is merged, I think it will be the best option.

jdexyz reviewed (approved):
Thank you for the PR !
Looks good to me, beside the console.error comment

edelciomolina committed:
Add: checkError

Ensure that in case it is really an error, then use console.error. Avoiding the output as an error in cases of warning that comes via stderr

edelciomolina committed:
Add: Check Dump With Verbose

edelciomolina committed:
Add: Tests for dump and restore with verbose args

edelciomolina commented:
@jdexyz All ready for merge.
- [x] Added function checkError
- [x] Added tests with verbose args
> before to approve, see https://github.com/EndoData/pg-dump-restore-node-wrapper/pull/5#issuecomment-2371512538. If pass there... here can closed without merge because is already updated there.
